### PR TITLE
wezterm: 20240203-110809-5046fc22 -> 20240203-110809-5046fc22-unstabl…

### PR DESCRIPTION
**Pull request by LamprosPitsillos**
…e-2024-05-20

## Description of changes
Wezterm does not start on `hyprland`
I believe this is the appropriate tracking [issue](https://github.com/wez/wezterm/issues/4483)

Updating to latest git fixed the issue, now wezterm starts as usual.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by LamprosPitsillos** [comment]
what to do about the long commit name ?

File: pkgs/applications/terminal-emulators/wezterm/default.nix [hunk 1]
```diff
@@ -37,9 +37,9 @@ rustPlatform.buildRustPackage rec {
   src = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/wezterm/default.nix:37]: We must adjust the version number, too.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Can we fetch a patch instead?

**Comment by SuperSandro2000** [comment]
> what to do about the long commit name ?

Nothing. That's just how it is.

File: pkgs/applications/terminal-emulators/wezterm/default.nix (hunk 1 above)
Comment (LamprosPitsillos) [reply pkgs/applications/terminal-emulators/wezterm/default.nix:37]: to the convention "version-unstable-date" i imagine ?

**Comment by LamprosPitsillos** [comment]
> Can we fetch a patch instead?

Wouldn't this require isolating the commit that solved the issue ?

File: pkgs/applications/terminal-emulators/wezterm/default.nix (hunk 1 above)
Comment (SuperSandro2000) [reply pkgs/applications/terminal-emulators/wezterm/default.nix:37]: hmmm, good question, I do not know... maybe the upstream format?

**Comment by SuperSandro2000** [comment]
> Wouldn't this require isolating the commit that solved the issue ?

yeah. Is that easily possible? I really don't want to fetch unstable as wezterm usually gets a bugfix release right after a bigger release.

**Comment by tlvenn** [comment]
It should be possible, the fix is trivial and in that commit:
> https://github.com/wez/wezterm/commit/1266912629ab951f7ee2924f4f8d949f564090c0

**Comment by SuperSandro2000** [comment]
Since I dislike tracking unstable software unless necessary and wezterm often needs a bug fix release a few days after its newest release, I couldn't convince myself to merge this PR.

Good news: wezterm has a flake and I recommend using that for the time being https://github.com/wez/wezterm/blob/main/nix/flake.nix

**Pull request closed without merging**